### PR TITLE
chore(docs): improve LATEST BY page

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Includes the following changes:
* Renames LATEST BY to LATEST ON everywhere in the docs. Keeps a redirect for the old `/latest-by` page address.
* Adds a clarifying text for the LATEST ON intro text.
* Adds a new section for LATEST ON over sub-query.

### CODE REVIEW

**Comment by bluestreak01** [comment]
I feel the opening paragraph could better explain why we can't just use `LIMIT -1`.

>LATEST ON PARTITION BY is used as part of a [SELECT statement](https://deploy-preview-730--unruffled-blackwell-31bfb2.netlify.app/docs/reference/sql/select/) for returning the most recent records per unique column value, commonly on STRING and SYMBOL column types. For the sake of brevity we to refer to this clause as LATEST ON.

I'd like to suggest something along the lines of:

```
For scenarios where multiple time series are stored in the same table it is relatively difficult to identify last items of these time series with standard SQL syntax. We introduces `LATEST ON` extension for `SELECT` clause to remove boilerplate clutter and splice table with relative ease. 
```

WDYT?

**Comment by bluestreak01** [comment]
Directly after syntax railroad diagram we should attempt to document syntax components. In similar way method args are documents along side what method does.

**Comment by bluestreak01** [comment]
Here:

image


Can we say "The old `LASTEST BY` syntax..." ?

**Comment by bluestreak01** [comment]
The document lacks description of how to time-slice, the last paragraph `LATEST ON first` does show mix of `where` clauses, but it is more powerful with time-slicing.

**Comment by bluestreak01** [comment]
We also should list all types, which can be used in `PARTITION BY` clause. I would also like to see description of how `LASTEST ON` will behave with non-symbol types and some advice around using timestamp to limit the performance impact

**Comment by puzpuzpuz** [comment]
> The document lacks description of how to time-slice, the last paragraph `LATEST ON first` does show mix of `where` clauses, but it is more powerful with time-slicing.

Addressed all comments except for this one in 73439d10637bfca7f875fa8300be66c15a03b651

**Comment by puzpuzpuz** [comment]
Addressed the remaining comment in 67ef3b743bd3da6ef3ca27151c6207a7cdc16ad3

File: docs/reference/sql/latest-on.md [hunk 1]
```diff
@@ -36,6 +51,16 @@ LATEST ON pickup_datetime PARTITION BY payment_type;
 | Cash         | 2019-06-30T23:59:54.000000Z | 2             |
 | Card         | 2019-06-30T23:59:56.000000Z | 1             |
 
+The above query returns the latest value within each time series stored in the
+table. Those time series are determined based on the column(s) specified in the
```
Comment (bluestreak01) [inline docs/reference/sql/latest-on.md:55]:
```suggestion
table. Those time series are determined based on the values in the column(s) specified in the
```

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: docs/reference/sql/latest-on.md [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
+  Reference documentation for using LATEST ON keywords with examples for
   illustration.
 ---
 
-`LATEST ON PARTITION BY` is used as part of a
-[SELECT statement](/docs/reference/sql/select/) for returning the most recent
-records per unique column value, commonly on `STRING` and `SYMBOL` column types.
-For the sake of brevity we to refer to this clause as `LATEST BY`.
+For scenarios where multiple time series are stored in the same table it is
+relatively difficult to identify the lastest items of these time series with
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply docs/reference/sql/latest-on.md]: No need for clarification, it's a typo. Thanks for spotting it. Fixed in 411eb87f12908c79fbcefdc59e3db3b97348f829

File: docs/reference/sql/latest-on.md (hunk 1 above)
Comment (puzpuzpuz) [reply docs/reference/sql/latest-on.md]: Applied in 411eb87f12908c79fbcefdc59e3db3b97348f829

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**